servers/tests/framework.rs

```diff
@@ -264,7 +264,8 @@ impl LocalServerContainer {
 		let _ = fs::create_dir_all(self.wallet_config.clone().data_file_dir);
 		let r = wallet::WalletSeed::init_file(&self.wallet_config, 32, "");
 
-		let client_n = HTTPNodeClient::new(&self.wallet_config.check_node_api_http_addr, None);
+		let client_n =
+			HTTPNodeClient::new(self.wallet_config.check_node_api_http_addr.clone(), None);
```

Comment:
what is the reason to pass `String`?

Reply:
Because `HTTPNodeClient::new` needs an ownership of this String,
before it received an `&str` and then cloned it using `to_owned()`, but this can cause unnecessary clone if the caller of the function doesn't need the String any more.
The more idiomatic thing to do is to try and not make any decisions for the caller, if you need ownership you should take it and let the owner decide if he should clone or not, and if you don't need ownership get a refrence.

Reply:
Looks like not beautiful as it was, adding a lot of `clone()` here and there.

Reply:
Hmm I agree that in some situations it can look less nice, 
but it can accumulate to a lot of memory if you pass a lot of `&Vec<T>` and `String` to functions and clone them inside and they can pass it on to another function that will clone it too etc.